lib/matplotlib/tests/test_figure.py

```diff
@@ -193,61 +192,27 @@ def test_gca():
         assert fig.add_axes() is None
 
     ax0 = fig.add_axes([0, 0, 1, 1])
-    with pytest.warns(
-            MatplotlibDeprecationWarning,
-            match=r'Calling gca\(\) with keyword arguments was deprecated'):
-        assert fig.gca(projection='rectilinear') is ax0
     assert fig.gca() is ax0
 
-    ax1 = fig.add_axes(rect=[0.1, 0.1, 0.8, 0.8])
-    with pytest.warns(
-            MatplotlibDeprecationWarning,
-            match=r'Calling gca\(\) with keyword arguments was deprecated'):
-        assert fig.gca(projection='rectilinear') is ax1
+    ax1 = fig.add_subplot(122)
```

Comment:
Why did `add_axes` with `rect=` go away?

Reply:
I believe these were testing whether `fig.gca(projection)` was working as expected, and we no longer allow the projection keyword argument, so I don't think we are gaining anything by keeping the addition of a rect axes here?

Reply:
Let's clarify what we are testing:
```suggestion
    # test that gca() picks up Axes created via add_subplot()
    ax1 = fig.add_subplot(122)
```

Also, 122 is a bit confusing because there is no obvious reason for it. Either use 111 or change the remaining `add_axes()` above to approximately fill the 121 space:  `add_axes([0, 0, 0.5, 1)`.